clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -953,6 +953,83 @@ test_regions_too_far()
     assert(status == scheduler_t::STATUS_ERROR_RANGE_INVALID);
 }
 
+static void
+test_regions_core_sharded()
+{
+    std::cerr << "\n----------------\nTesting region on core-sharded-on-disk trace\n";
+    static constexpr memref_tid_t TID_A = 42;
+    static constexpr memref_tid_t TID_B = 99;
+    static constexpr addr_t PC_POST_FOOTER = 101;
+    std::vector<trace_entry_t> memrefs = {
+        /* clang-format off */
+        make_thread(TID_A),
+        make_pid(1),
+        make_marker(TRACE_MARKER_TYPE_PAGE_SIZE, 4096),
+        make_timestamp(10),
+        make_marker(TRACE_MARKER_TYPE_CPU_ID, 1),
+        make_instr(1),
+        make_instr(2),
+        make_exit(TID_A),
+        // Test skipping across a footer.
+        make_footer(),
+        make_thread(TID_B),
+        make_pid(1),
+        make_marker(TRACE_MARKER_TYPE_PAGE_SIZE, 4096),
+        make_timestamp(10),
+        make_marker(TRACE_MARKER_TYPE_CPU_ID, 1),
+        make_instr(PC_POST_FOOTER),
+        make_instr(PC_POST_FOOTER + 1),
+        make_exit(TID_B),
+        make_footer(),
+        /* clang-format on */
+    };
+    std::vector<scheduler_t::input_reader_t> readers;
+    readers.emplace_back(std::unique_ptr<mock_reader_t>(new mock_reader_t(memrefs)),
+                         std::unique_ptr<mock_reader_t>(new mock_reader_t()), 1);
+
+    std::vector<scheduler_t::range_t> regions;
+    // Start beyond the footer.
+    regions.emplace_back(3, 0);
```

Comment:
Curious: does this test fail without the fix only when we have ROI-induced skip? Does it also fail if `regions` is not specified?

Reply:
It does not fail if regions is not there: it only happens with ROI/skipping.